### DISCUSSION TITLE
Is it possible to finely control whether a specific demand order is split?

### DISCUSSION THREAD

**Discussion by onepiecewyz** [frePPLe/frepple#638] [General]
Currently, FrePPLe controls the "allowsplits" parameter to specify whether global automatic order splitting is enabled. Is it possible to finely control whether a specific demand order is split? For example, with "allowsplits=true," can we specify that a particular demand order should not be split automatically?

**Comment by jdetaeye** [comment]
This is possible with a custom app.

We recently used a similar customization for a customer that wanted to plan some sales orders in "forward planning mode / ASAP delivery" and others in "backward planning mode / using the due date".
This was done by adjusting the global setting "administratitive lead time" for individual demands.

A custom app with the following content in a file "yourapp/commands.py" will achieve this.  See https://frepple.com/docs/current/developer-guide/creating-an-extension-app.php for full details an extension apps.

```
import logging
import os

from django.db import connections, DEFAULT_DB_ALIAS


from freppledb.common.commands import PlanTaskRegistry, PlanTask
from freppledb.execute.commands import SupplyPlanning


@PlanTaskRegistry.register
class AsphericonSupplyPlanning(SupplyPlanning):

    default_administrativeleadtime = None

    @classmethod
    def debugDemand(cls, dem, mode):
        import frepple

        if cls.default_administrativeleadtime is None:
            cls.default_administrativeleadtime = cls.solver.administrativeleadtime
        if getattr(dem, "asap", False):
            # This demand is flagged with the "asap" attribute. We plan its delivery asap
            # by dynamically adjusting the administrativeleadtime
            cls.solver.administrativeleadtime = max(
                (dem.due - frepple.settings.current).total_seconds(), 0
            )
            if cls.solver.loglevel > 0:
                print("Switching to asap mode for %s" % dem.name)
        else:
            cls.solver.administrativeleadtime = cls.default_administrativeleadtime

    @classmethod
    def run(cls, database=DEFAULT_DB_ALIAS, **kwargs):
        SupplyPlanning.debugDemand = cls.debugDemand
        SupplyPlanning.run(database=database, **kwargs)
```
